src/borg/testsuite/archiver/corruption.py

```diff
@@ -46,13 +46,16 @@ def corrupt(file, amount=1):
 def test_cache_chunks(archiver):
     corrupt_archiver(archiver)
     create_src_archive(archiver, "test")
-    corrupt(os.path.join(archiver.cache_path, "chunks"))
+    chunks_path = os.path.join(archiver.cache_path, "chunks")
+    chunks_before_corruption = set(ChunkIndex(path=chunks_path).iteritems())
+    corrupt(chunks_path)
     if archiver.FORK_DEFAULT:
         out = cmd(archiver, "rinfo")
     else:
         out = cmd(archiver, "rinfo")
     assert "forcing a cache rebuild" in out
-
+    chunks_after_repair = set(ChunkIndex(path=chunks_path).iteritems())
+    assert chunks_after_repair == chunks_before_corruption
```

Comment:
Looks good as far as the on-disk index is concerned.

Not sure about the in-memory index that is actually used in the borg invocation when the rebuild happens.

Reply:
New commit with the in-memory index checked (inspired by other test files)